src/foundry/common/documents/tile.mjs.d.ts

```diff
@@ -0,0 +1,204 @@
+// FOUNDRY_VERSION: 10.291
+
+import Document, { DocumentMetadata } from "../abstract/document.mjs";
+import * as CONST from "../constants.mjs";
+import * as fields from "../data/fields.mjs";
+import { TextureData } from "../data/data.mjs";
+
+declare global {
+  type TileData = BaseTile.Properties;
+
+  type TileOcclusionData = BaseTile.Properties["occlusion"];
+
+  type TileVideoData = BaseTile.Properties["video"];
+}
+
+/**
+ * The Document definition for a Tile.
+ * Defines the DataSchema and common behaviors for a Tile which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseTile extends BaseTile.Properties {}
+declare class BaseTile extends Document<BaseTile.SchemaField, BaseTile.Metadata> {
+  /**
+   * @param data    - Initial data from which to construct the Tile
+   * @param context - Construction context options
+   */
+  constructor(data: BaseTile.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseTile.Metadata>;
+
+  static override defineSchema(): BaseTile.Schema;
+
+  static override migrateData(source: object): object;
+
+  static override shimData(
+    data: object,
+    {
+      embedded
+    }?: {
+      /**
+       * Apply shims to embedded models?
+       * @defaultValue `true`
+       */
+      embedded?: boolean;
+    }
+  ): object;
+}
+export default BaseTile;
+
+declare namespace BaseTile {
+  type Metadata = Merge<
+    DocumentMetadata,
+    { name: "Tile"; collection: "tiles"; label: "DOCUMENT.Tile"; labelPlural: "DOCUMENT.Tiles" }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
+  type UpdateData = fields.SchemaField.InnerAssignmentType<Schema>;
+  type Properties = fields.SchemaField.InnerInitializedType<Schema>;
+  type Source = fields.SchemaField.InnerPersistedType<Schema>;
+
+  interface Schema extends DataSchema {
+    /**
+     * The _id which uniquely identifies this Tile embedded document
+     * @defaultValue `null`
+     */
+    _id: fields.DocumentIdField;
+
+    /**
+     * An image or video texture which this tile displays.
+     * @defaultValue `null`
+     */
+    texture: TextureData<{}, { categories: ["IMAGE", "VIDEO"]; initial: null; wildcard: false }>;
```

Comment:
Can you check what the type of `categories` comes down to? I'm not sure if we need to use `("IMAGE" | "VIDEO")[]` instead. This might come out as a tuple in the end, where the other notation allows for an array of any order.

Reply:
Reading the source code, it looks like order is not supposed to be important, and you should be allowed to specify 0 to many, rather than a consistent tuple of values.

I'll make the type change to this line.